logstash-core/api/lib/app/system/info_command.rb

```diff
@@ -4,20 +4,30 @@
 class LogStash::Api::SystemInfoCommand < LogStash::Api::Command
 
   def run
-    report = { "version"   => "1.0.0",
-               "host_name" => "foobar",
-               "plugins"   => { "count" => 10, "list" => [] }
+    plugins_list = available_plugins
+    {
+      "version"   => LOGSTASH_CORE_VERSION,
+      "hostname" => hostname,
+      "plugins"   => { "count" => plugins_list.count, "list" => plugins_list.sort }
     }
-    report
   end
 
   private
 
-  def type
-    ["input", "output", "filter"][rand(3)]
+  def hostname
+    `hostname`.strip
   end
```

Comment:
When I disconnect my laptop and I try it, I don't have have any errors and my hostname is also invalid.
```ruby
~/e/l/logstash-input-beats git:fix/timing-issue-with-test ❯❯❯ irb                                                                                                                                                                   ◼
jruby-1.7.23 :001 > require 'Socket'
 => true
jruby-1.7.23 :002 > Socket.gethostname
 => "sashimi.local"
jruby-1.7.23 :003 >